server/container_create.go

```diff
@@ -471,14 +471,22 @@ func (s *Server) createSandboxContainer(ctx context.Context, containerID string,
 			specgen.SetProcessOOMScoreAdj(int(oomScoreAdj))
 		}
 
+		var cgPath string
+		scopePrefix := "crio"
+		parent := "/crio"
+		useSystemd := s.config.CgroupManager == "systemd"
+		if useSystemd {
+			parent = "system.slice"
+		}
 		if sb.CgroupParent() != "" {
-			if s.config.CgroupManager == "systemd" {
-				cgPath := sb.CgroupParent() + ":" + "crio" + ":" + containerID
-				specgen.SetLinuxCgroupsPath(cgPath)
-			} else {
-				specgen.SetLinuxCgroupsPath(sb.CgroupParent() + "/" + containerID)
-			}
+			parent = sb.CgroupParent()
+		}
+		if useSystemd {
+			cgPath = parent + ":" + scopePrefix + ":" + containerID
+		} else {
+			cgPath = filepath.Join(parent, containerID)
```

Comment:
@derekwaynecarr let me know if you want `crio-` prefix with `containerID` here (and in the systemd case), otherwise, current code is what docker does today and _should_ probably be ok

Reply:
FWIW, I'd vote for doing what docker does today.

Reply:
Actually we probably need to prefix containerID as if parent comes from kube, it's probably not "crio-" so better prefix it. This should not be need in the systemd case where scopePrefix is crio

Reply:
This is fixed now and cgroupfs syntax is now `crio-$CONTAINER_ID` as requested.